src/spikeinterface/core/template_tools.py

```diff
@@ -31,7 +31,8 @@ def get_dense_templates_array(one_object: Templates | SortingAnalyzer, return_sc
             )
         ext = one_object.get_extension("templates")
         if ext is not None:
-            templates_array = ext.data["average"]
+            templates_array = ext.data.get("average") or ext.data.get("median")
+            assert templates_array is not None, "Average or median templates have not been computed."
```

Comment:
```suggestion
            if "average" in ext.data:
                templates_array = ext.data.get("average")
            elif "median" in ext.data:
                templates_array = ext.data.get("median")
            else:
                raise ValueError("Average or median templates have not been computed.")
```

@OlivierPeron let's refactor this way since the current implementation is throwing an error